### PR TITLE
Support Xcode 12 by excluding arm64 simulators and bumping Telemetry to 0.10.4

### PR DESCRIPTION
**Pull request by nagineni**
~Carthage build fix for testing.~

Preparation for release 6.2.0. This PR:
- Bumps `mapbox-events-ios` to 0.10.4
- Supports Xcode 12 by excluding arm64 simulators. This helps with when using Carthage for dependency management (See https://github.com/Carthage/Carthage/issues/3019)
- Fixes an issue where the map's `ConstrainMode` was not being set.

### CODE REVIEW

**Review by neelmistry94 — APPROVED** [review]
Pulled your branch and ran make iproj and make iOS and everything built as expected

**Comment by nagineni** [comment]
@neelmistry94 thanks for the verification.

File: platform/ios/CHANGELOG.md [hunk 1]
```diff
@@ -29,7 +35,7 @@ Mapbox welcomes participation and contributions from everyone. Please read [CONT
 ### 🔧 Dependencies
 
 * Core library updated to `5.0.1`. ([#438](https://github.com/mapbox/mapbox-gl-native-ios/pull/453))
-* `mapbox-events-ios` versioning updated to `~> 0.10.2`.
+* `mapbox-events-ios` versioning updated to `~> 0.10.3`.
```
Comment (knov) [inline platform/ios/CHANGELOG.md:38]: Please update

Comment (julianrex) [reply platform/ios/CHANGELOG.md]: Resolved in later commit.

**Review by fabian-guerra — APPROVED** [review]
Tested using Xcode 11.7 and Xcode12 GA 209 and works fine. Thank you!

**Pull request merged**